pkg/controller/snapshot_controller.go

```diff
@@ -751,36 +752,30 @@ func (ctrl *csiSnapshotController) deleteSnapshotContentOperation(content *crdv1
 
 	// get secrets if VolumeSnapshotClass specifies it
 	var snapshotterCredentials map[string]string
-	var err error
-
-	// Check if annotation exists
-	if metav1.HasAnnotation(content.ObjectMeta, AnnDeletionSecretRefName) && metav1.HasAnnotation(content.ObjectMeta, AnnDeletionSecretRefNamespace) {
-		annDeletionSecretName := content.Annotations[AnnDeletionSecretRefName]
-		annDeletionSecretNamespace := content.Annotations[AnnDeletionSecretRefNamespace]
-
-		snapshotterSecretRef := &v1.SecretReference{}
-
-		if annDeletionSecretName == "" || annDeletionSecretNamespace == "" {
-			return fmt.Errorf("cannot delete snapshot %#v, err: secret name or namespace not specified", content.Name)
-		}
-
-		snapshotterSecretRef.Name = annDeletionSecretName
-		snapshotterSecretRef.Namespace = annDeletionSecretNamespace
-
-		snapshotterCredentials, err = getCredentials(ctrl.client, snapshotterSecretRef)
-		if err != nil {
-			// Continue with deletion, as the secret may have already been deleted.
-			klog.Errorf("Failed to get credentials for snapshot %s: %s", content.Name, err.Error())
-		}
-	}
+	/* TODO(@Xing-yang): secrete ref retrivial needs to be implemented here
+	  	snapshotClassName := content.Spec.VolumeSnapshotClassName
+		if snapshotClassName != nil {
+			if snapshotClass, err := ctrl.classLister.Get(*snapshotClassName); err == nil {
+				// Resolve snapshotting secret credentials.
+				// No VolumeSnapshot is provided when resolving delete secret names, since the VolumeSnapshot may or may not exist at delete time.
+				snapshotterSecretRef, err := getSecretReference(snapshotClass.Parameters, content.Name, nil)
+				if err != nil {
+					return err
+				}
+				snapshotterCredentials, err = getCredentials(ctrl.client, snapshotterSecretRef)
+				if err != nil {
+					return err
+				}
+			}
+		}*/
```

Comment:
Note that the secrets PR was merged: https://github.com/kubernetes-csi/external-snapshotter/pull/165.  Part of that PR is missing here.

Reply:
per discussion, it seems there was some merge conflict resolved wrongly in previous commits. we will address this prior to final merge